test/controllers/ApplicationControllerSpec.scala

```diff
@@ -385,6 +387,85 @@ class ApplicationControllerSpec
       }
     }
 
+    "manageGrantLengthPage" should {
+      "return the manage grant length page for an admin" in new Setup {
+        givenTheGKUserIsAuthorisedAndIsAnAdmin()
+        givenTheAppWillBeReturned()
+
+        val result = underTest.manageGrantLength(applicationId)(anAdminLoggedInRequest)
+
+        status(result) shouldBe OK
+        contentAsString(result) should include("Manage application grant length")
+      }
+
+      "return the manage grant length page for a super user" in new Setup {
+        givenTheGKUserIsAuthorisedAndIsASuperUser()
+        givenTheAppWillBeReturned()
+
+        val result = underTest.manageGrantLength(applicationId)(aSuperUserLoggedInRequest)
+
+        status(result) shouldBe OK
+        contentAsString(result) should include("Manage application grant length")
+      }
+
+      "return the forbidden page for a normal user" in new Setup {
+        givenTheGKUserHasInsufficientEnrolments()
+        givenTheAppWillBeReturned()
+
+        val result = underTest.manageGrantLength(applicationId)(aLoggedInRequest)
+
+        status(result) shouldBe FORBIDDEN
+        contentAsString(result) should include("You do not have permission")
+      }
+    }
+
+    "updateGrantLength" should {
+      "call the service to update the grant length when a valid form is submitted for an admin" in new Setup {
+        givenTheGKUserIsAuthorisedAndIsAnAdmin()
+        givenTheAppWillBeReturned()
+
+        ApplicationServiceMock.UpdateGrantLength.succeeds()
+
+        val request = anAdminLoggedInRequest.withFormUrlEncodedBody("grantLength" -> "547")
+
+        val result = addToken(underTest.updateGrantLength(applicationId))(request)
+
+        status(result) shouldBe OK
+
+        verify(mockApplicationService).updateGrantLength(eqTo(basicApplication), eqTo(GrantLength.EIGHTEEN_MONTHS))(*)
+        verify(underTest.authConnector).authorise(eqTo(Enrolment(adminRole)), *)(*, *)
+        verifyAuthConnectorCalledForAdmin
+      }
+
+      "return a bad request when an invalid form is submitted for an admin user" in new Setup {
+        givenTheGKUserIsAuthorisedAndIsAnAdmin()
+        givenTheAppWillBeReturned()
+
+        val request = anAdminLoggedInRequest.withFormUrlEncodedBody()
+
+        val result = addToken(underTest.updateGrantLength(applicationId))(request)
+
+        status(result) shouldBe BAD_REQUEST
+
+        verify(mockApplicationService, never).updateGrantLength(*, *)(*)
```

Comment:
If this compiles and runs then it's using the Java version of verify.  Any "extends AnyVal" (such as applicationId) needs to use *[ApplicationId] as the matcher.

Reply:
Ok - so I'm really surprised you've had no issues.  The * don't match any AnyVal types.  Perhaps with it being a never it does not matter.  I'd still be tempted to change the * though to *[ApplicationId] because if anyone takes this line to use for a "once" it wouldn't work.  I know it's a strange issue with AnyVals but I've been bitten enough times I'm paranoid about it.